scripts/pr-add-size-label.sh

```diff
@@ -0,0 +1,396 @@
+# should be applied to a PR.
+#
+# key: Label suffix to add to PR if it matches the "size" specified
+#  by the value.
+# value: Size range.
+size_ranges=(
+    [tiny]="$KATA_PR_SIZE_RANGE_TINY"
+    [small]="$KATA_PR_SIZE_RANGE_SMALL"
+    [medium]="$KATA_PR_SIZE_RANGE_MEDIUM"
+    [large]="$KATA_PR_SIZE_RANGE_LARGE"
+    [huge]="$KATA_PR_SIZE_RANGE_HUGE"
+)
+
+info()
+{
+    echo "INFO: $*"
+}
+
+die()
+{
+    echo >&2 "ERROR: $*"
+    exit 1
+}
+
+setup()
+{
+    local cmds=()
+
+    cmds+=("diffstat")
+    cmds+=("filterdiff")
+
+    # https://github.com/cli/cli
+    cmds+=("gh")
+
+    local cmd
+
+    local ret
+
+    for cmd in "${cmds[@]}"
+    do
+        { command -v "$cmd" &>/dev/null; ret=$?; } || true
+        [ "$ret" -eq 0 ] || die "need command '$cmd'"
+    done
+
+    local vars=()
+
+    vars+=("GITHUB_TOKEN")
+
+    local var
+
+    for var in "${vars[@]}"
+    do
+        local value
+
+        value=$(printenv "$var" || true)
+
+        [ -n "${value}" ] || die "need to set '$var'"
+    done
+
+    # Force non interactive mode
+    gh config set prompt disabled
+}
+
+# Return an integer representing the "size" of a PR,
+# calculated as the sum of additions and deletions
+# (the amount of change).
+#
+# Note: This function attempts to exclude:
+#
+# - Vendored code changes (pristine upstream files) since vendored code,
+#   by definition, should be treated as "read only" and those changes
+#   result as a _side effect_ of the actual change (which tends to be a
+#   version bump in a configuration file (which causes the pristine vendor
+#   code to be updated).
+#
+# - Auto-generated code.
+#
+# - generated lock/status files for languages like golang and rust.
+get_pr_size()
+{
+    local pr="${1:-}"
+    [ -z "$pr" ] && die "need PR number"
+
+    local stats
+
+    # Determine the amount of change for the specified PR.
+    #
+    # Notes:
+    #
+    # - The exclusions specified here should be similar to the logic in:
+    #
+    #   https://github.com/kata-containers/tests/blob/main/.ci/static-checks.sh
+    #
+    # - Example output showing the diffstat(1) formats:
+    #
+    #     "99 files changed, 12345 insertions(+), 987 deletions(-)"
+    #     "1 file changed, 1 insertion(+)"
+    #     "1 file changed, 2 deletions(-)"
+    stats=$(gh pr diff "$pr" |\
+            filterdiff \
+                --exclude='Cargo.lock' \
+                --exclude='go.mod' \
+                --exclude='go.sum' \
+                --exclude='*.pb.go' \
+                --exclude='*.pb_test.go' \
+                --exclude='*/src/libs/protocols/src/*.rs' \
+                --exclude='*/vendor/*' \
+                --exclude='*/virtcontainers/pkg/cloud-hypervisor/client' \
+                --exclude='*/virtcontainers/pkg/firecracker/client' \
+                |\
+            diffstat -s)
+
+    local additions
+    local deletions
+
+    additions=$(echo "$stats" |\
+        grep -Eo '[0-9][0-9]* insertions?' |\
```

Comment:
Is it safe to use `[0-9]+` instead of `[0-9][0-9]*` ?

Reply:
Agreed - I've simplified all instances of that pattern.